### PR TITLE
[EWPP-1424] EWPP-1426: Provide a block that renders the DFF webtools widget.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: modules/oe_webtools_page_feedback/oe_webtools_page_feedback.module [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+declare(strict_types = 1);
+
+use Drupal\Core\Cache\CacheableMetadata;
+
+/**
+ * Implements hook_page_attachments().
+ *
+ * Attach smart loader.
+ */
+function oe_webtools_page_feedback_page_attachments(array &$attachments): void {
```
Comment (upchuk) [inline modules/oe_webtools_page_feedback/oe_webtools_page_feedback.module:17]: Why you attaching this here? Just make add it to the the render array in the block where we render the actual thing...like SocialShareBlock. Which in fact ..you do!

File: modules/oe_webtools_page_feedback/src/EventSubscriber/PageFeedbackConfigCacheInvalidator.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  /**
+   * Invalidate cache tags when page feedback form config object changes.
+   *
+   * @param \Drupal\Core\Config\ConfigCrudEvent $event
+   *   The event to process.
+   */
+  public function onChange(ConfigCrudEvent $event) {
+    if ($event->getConfig()->getName() === 'oe_webtools_page_feedback.settings') {
+      $this->cacheTagsInvalidator->invalidateTags(['library_info', 'rendered']);
```
Comment (upchuk) [inline modules/oe_webtools_page_feedback/src/EventSubscriber/PageFeedbackConfigCacheInvalidator.php:42]: Why is this needed? We should just add the cache tags on the block render array. Saving the config should not add/remove a library from the registry so I fail to see why this is needed.

File: modules/oe_webtools_page_feedback/src/Plugin/Block/PageFeedbackFormBlock.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(): array {
+    $feedback_form_json = [
+      'service' => 'dff',
+      'id' => $this->configFactory->get('oe_webtools_page_feedback.settings')->get('feedback_form_id'),
+      'lang' => $this->languageManager->getCurrentLanguage()->getId(),
```
Comment (upchuk) [inline modules/oe_webtools_page_feedback/src/Plugin/Block/PageFeedbackFormBlock.php:94]: Does this work with PT? As far as I know, they expect PT, not pt-pt.

File: modules/oe_webtools_page_feedback/tests/src/FunctionalJavascript/PageFeedbackFormTest.php [hunk 4]
```diff
@@ -24,9 +28,23 @@ class PageFeedbackFormTest extends WebDriverTestBase {
...
+    parent::setUp();
+
+    $this->drupalPlaceBlock('oe_webtools_page_feedback_form');
+    $this->drupalPlaceBlock('page_title_block');
+    $this->drupalCreateContentType(['type' => 'page', 'name' => 'Basic page']);
+    ConfigurableLanguage::createFromLangcode('pt')->save();
+  }
+
+  /**
+   * Tests Page Feedback Form configuration and block rendering.
```
Comment (upchuk) [inline modules/oe_webtools_page_feedback/tests/src/FunctionalJavascript/PageFeedbackFormTest.php:43]: Then the class should also be renamed no?

File: modules/oe_webtools_page_feedback/tests/src/FunctionalJavascript/PageFeedbackFormTest.php [hunk 5]
```diff
@@ -24,9 +28,23 @@ class PageFeedbackFormTest extends WebDriverTestBase {
...
-   * Tests Page Feedback Form configuration.
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->drupalPlaceBlock('oe_webtools_page_feedback_form');
+    $this->drupalPlaceBlock('page_title_block');
+    $this->drupalCreateContentType(['type' => 'page', 'name' => 'Basic page']);
+    ConfigurableLanguage::createFromLangcode('pt')->save();
```
Comment (upchuk) [inline modules/oe_webtools_page_feedback/tests/src/FunctionalJavascript/PageFeedbackFormTest.php:39]: What language is `pt`? Because the Drupal language code for Portuguese is `pt-pt`.

**Review by upchuk — CHANGES_REQUESTED** [review]

File: modules/oe_webtools_page_feedback/oe_webtools_page_feedback.module (hunk 1 above)
Comment (sergepavle) [reply modules/oe_webtools_page_feedback/oe_webtools_page_feedback.module]: Fixed.

File: modules/oe_webtools_page_feedback/src/EventSubscriber/PageFeedbackConfigCacheInvalidator.php (hunk 2 above)
Comment (sergepavle) [reply modules/oe_webtools_page_feedback/src/EventSubscriber/PageFeedbackConfigCacheInvalidator.php]: Removed.

File: modules/oe_webtools_page_feedback/src/Plugin/Block/PageFeedbackFormBlock.php (hunk 3 above)
Comment (sergepavle) [reply modules/oe_webtools_page_feedback/src/Plugin/Block/PageFeedbackFormBlock.php]: Webtools don't support pt-pt lang code. Seems like the issue is related not only to this widget but also to other submodules. Should we do a follow-up regarding these issues?

File: modules/oe_webtools_page_feedback/tests/src/FunctionalJavascript/PageFeedbackFormTest.php (hunk 4 above)
Comment (sergepavle) [reply modules/oe_webtools_page_feedback/tests/src/FunctionalJavascript/PageFeedbackFormTest.php:43]: done

File: modules/oe_webtools_page_feedback/tests/src/FunctionalJavascript/PageFeedbackFormTest.php (hunk 5 above)
Comment (sergepavle) [reply modules/oe_webtools_page_feedback/tests/src/FunctionalJavascript/PageFeedbackFormTest.php]: fixed

**Review by upchuk — CHANGES_REQUESTED** [review]
